### PR TITLE
Add sub to regex.

### PR DESCRIPTION
**Pull request by yuyantingzero**

### CODE REVIEW

File: perfkitbenchmarker/regex_util.py [hunk 1]
```diff
@@ -74,3 +74,26 @@ def ExtractAllMatches(regex, text):
...
+    pattern: string. Pattern to be replaced.
+    repl: string. Replacement pattern.
+    text: string. Text to search.
+  Returns:
+    A string after replacing all patterns with repl.
+  Raises:
+    NoMatchError: when 'pattern' isn't found in string.
+  """
+  match = re.search(pattern, text)
+  print match
```
Comment (cmccoy) [inline perfkitbenchmarker/regex_util.py]: Please remove.

File: perfkitbenchmarker/regex_util.py [hunk 2]
```diff
@@ -74,3 +74,26 @@ def ExtractAllMatches(regex, text):
...
+
+  Example:
+  >>> re.sub(r'<(\w+)>', r'[\1]', 'foo <bar> <foo> bar')
+  'foo [bar] [foo] bar'
+  Args:
+    pattern: string. Pattern to be replaced.
+    repl: string. Replacement pattern.
+    text: string. Text to search.
+  Returns:
+    A string after replacing all patterns with repl.
```
Comment (cmccoy) [inline perfkitbenchmarker/regex_util.py]: Nit: 'text' after replacing all matches of 'pattern' with 'repl'

File: perfkitbenchmarker/regex_util.py [hunk 3]
```diff
@@ -74,3 +74,26 @@ def ExtractAllMatches(regex, text):
     raise NoMatchError('No match for pattern "{0}" in "{1}"'.format(
         regex, text))
   return match
+
+
+def Sub(pattern, repl, text):
+  """Substitute all patterns in string with repl pattern.
+
+  Example:
+  >>> re.sub(r'<(\w+)>', r'[\1]', 'foo <bar> <foo> bar')
```
Comment (cmccoy) [inline perfkitbenchmarker/regex_util.py]: Why have an example of `re.sub` here?

File: perfkitbenchmarker/regex_util.py [hunk 4]
```diff
@@ -74,3 +74,26 @@ def ExtractAllMatches(regex, text):
...
+    repl: string. Replacement pattern.
+    text: string. Text to search.
+  Returns:
+    A string after replacing all patterns with repl.
+  Raises:
+    NoMatchError: when 'pattern' isn't found in string.
+  """
+  match = re.search(pattern, text)
+  print match
+  if not match:
```
Comment (cmccoy) [inline perfkitbenchmarker/regex_util.py]: Nit: if you're not using `match`, consider dropping the local.

**Comment by yuyantingzero** [comment]
PTAL.

File: perfkitbenchmarker/regex_util.py [hunk 5]
```diff
@@ -74,3 +74,21 @@ def ExtractAllMatches(regex, text):
     raise NoMatchError('No match for pattern "{0}" in "{1}"'.format(
         regex, text))
   return match
+
+
+def Sub(pattern, repl, text):
```
Comment (cmccoy) [inline perfkitbenchmarker/regex_util.py]: Consider naming this 'Substitute'.

**Comment by cmccoy** [comment]
One other small comment. Otherwise LGTM.

**Pull request merged**